src/ApiManagement/ApiManagement.ServiceManagement/ApiManagementClient.cs

```diff
@@ -2238,6 +2238,54 @@ public void LoggerRemove(PsApiManagementContext context, string loggerId)
                 loggerUpdateParameters,
                 "*");
         }
+
+        public PsApiManagementLogger AttachApiDiagnosticLogger(PsApiManagementContext context,
+                                string apiId,
+                                string loggerId,
+                                string diagnosticId)
+        {
+            DiagnosticContract diagnosticParam = new DiagnosticContract(true);
+            var diagnostic = Client.ApiDiagnostic.CreateOrUpdate(context.ResourceGroupName,
+                                                      context.ServiceName,
+                                                      apiId,
+                                                      diagnosticId,
+                                                      diagnosticParam
+                                                      );
+           
+            var loggerContract = Client.ApiDiagnosticLogger.CreateOrUpdate(context.ResourceGroupName,
+                                                      context.ServiceName,
+                                                      apiId,
+                                                      diagnosticId,
+                                                      loggerId);
+            return Mapper.Map<PsApiManagementLogger>(loggerContract);
+
+        }
+
+        public void DisableDiagnosticApi(PsApiManagementContext context,
+                        string apiId,
+                        string diagnosticId)
+        {
+            var ifmatch = "*";
+            Client.ApiDiagnostic.Delete(context.ResourceGroupName,
+                                                      context.ServiceName,
+                                                      apiId,
+                                                      diagnosticId,
+                                                      ifmatch
+                                                      );
+
+    
+        }
```

Comment:
@solankisamir do we have cmdlets for the global-level diagnostics? Why can't we just copy them here and add Api Id parameter? Am I missing something here?